preprocessing_scripts/ert_rft.py

```diff
@@ -0,0 +1,196 @@
+        columns=[
+            "well",
+            "time",
+            "obs",
+            "error",
+            "utm_x",
+            "utm_y",
+            "measured_depth",
+            "true_vertical_depth",
+            "zone",
+        ],
+    )
+
+    df["obs"] = pd.to_numeric(df["obs"])
+    df["error"] = pd.to_numeric(df["error"])
+    df["utm_x"] = pd.to_numeric(df["utm_x"])
+    df["utm_y"] = pd.to_numeric(df["utm_y"])
+    df["measured_depth"] = pd.to_numeric(df["measured_depth"])
+    df["true_vertical_depth"] = pd.to_numeric(df["true_vertical_depth"])
+
+    df = df.astype({"time": str})
+    df["time"] = df["time"].apply(lambda x: f"{x[0:4]}-{x[4:6]}-{x[6:]}")
+    df["REPORTSTEP"] = reportstep
+    return df
+
+
+######################################################################
+# df_rft_addsim: this function will make a merged dataframe of obsdata and simdata
+# step 2
+######################################################################
+
+
+def df_rft_addsim(args, verbose=True):
+
+    runpath = args["realization"].runpath()
+    print(f"Working on realization {args['realization'].index}")
+    obsdf = args["obsdf"]
+    
+    dfs = []
+    for well, welldf in obsdf.groupby("well"):
+        
+        for reportstep, rsdf in welldf.groupby("REPORTSTEP"):
+            
+            with open(
+                os.path.join(runpath, f"RFT_{well}_{reportstep}"), "r"
+            ) as rftfile:
+                sim_rft = []
+                sim_active = []
+                for line in rftfile:
+                    pressure = line.split()[0]
+                    active = 1
+                    if pressure == "-1":
+                        pressure = 0
+                        active = 0
+                    sim_rft.append(pressure)
+                    sim_active.append(active)
+                df = rsdf.copy()
+                df["pressure"] = sim_rft
+                df["isactive"] = sim_active
+                dfs.append(df)
+        
+
+    return pd.concat(dfs)
+
+
+reportsteps = [
+    {
+        "folder": "../observed_data/ert_rft/",
+        "well_date": "well_date_rft.txt",
+        "reportstep": 1,
+    },
+    # {
+    #     "folder": "put another folder here",
+    #     "well_date": "well_date_plt.txt",
+    #     "reportstep": 2,
+    # }
+
+]
+
+obsdf = get_rft_reportsteps(reportsteps)
+
+ens = [
+    ScratchEnsemble(enspath[0], enspath[1])
+    for enspath in [
+        [
+            "iter-0", "../reek_history_match/realization-*/iter-0"
+        ],
+        [
+            "iter-1", "../reek_history_match/realization-*/iter-1"
+        ],
+        [
+            "iter-2", "../reek_history_match/realization-*/iter-2"
+        ],
+        [
+            "iter-3", "../reek_history_match/realization-*/iter-3"
+        ],
+    ]
+]
+ens_filtered = [e.filter("OK") for e in ens]
+ensset = EnsembleSet("", ens_filtered)
+
+df = ensset.apply(df_rft_addsim, obsdf=obsdf)
+df.to_csv("rft_ert.csv", index=False)
```

Comment:
Maybe not mix in reek-specific config? Rather take reportsteps and ensemble names+paths as input, and configure/run it from a separate file? So that users don't have to edit the main script?

Also: consider to use report_steps instead of reportsteps, as that is the naming in ERT? (though ERT has it in capital letters)